runtime/src/asset.rs

```diff
@@ -1072,6 +1239,91 @@ impl<T: Trait> Module<T> {
         token.owner_did == did
     }
 
+    pub fn is_ticker_available(ticker: &Vec<u8>) -> bool {
+        // Assumes uppercase ticker
+        if <Tickers<T>>::exists(ticker.clone()) {
+            let now = <timestamp::Module<T>>::get();
+            if let Some(expiry) = Self::ticker_registration(ticker.clone()).expiry {
+                if now <= expiry {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    pub fn is_ticker_registry_valid(ticker: &Vec<u8>, did: IdentityId) -> bool {
+        // Assumes uppercase ticker
+        if <Tickers<T>>::exists(ticker.clone()) {
+            let now = <timestamp::Module<T>>::get();
+            let ticker_reg = Self::ticker_registration(ticker.clone());
+            if ticker_reg.owner == did {
+                if let Some(expiry) = ticker_reg.expiry {
+                    if now > expiry {
+                        return false;
+                    }
+                } else {
+                    return true;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// Returns 0 if ticker is registered to someone else
+    /// 1 if ticker is available for registry
+    /// 2 if ticker is already registered to provided did
+    pub fn is_ticker_available_or_registered_to(ticker: &Vec<u8>, did: IdentityId) -> u8 {
```

Comment:
Use an enum (like `enum TickerRegistration( RegisteredByOther, Available, RegisteredByDid)` instead of `u8` on return.